harness/determined/common/storage/azure.py

```diff
@@ -30,73 +31,69 @@ def __init__(
+            logging.info(f"Uploading checkpoint {uuid} to Azure Blob Storage.")
+            self.upload(uuid, storage_dir)
         finally:
-            self._remove_checkpoint_directory(metadata.storage_id)
+            self._remove_checkpoint_directory(uuid)
 
     @contextlib.contextmanager
-    def restore_path(self, metadata: StorageMetadata) -> Iterator[str]:
-        storage_dir = os.path.join(self._base_path, metadata.storage_id)
+    def restore_path(self, uuid: str) -> Iterator[str]:
+        storage_dir = os.path.join(self._base_path, uuid)
         os.makedirs(storage_dir, exist_ok=True)
 
-        logging.info(
-            "Downloading checkpoint {} from Azure Blob Storage".format(metadata.storage_id)
-        )
-        self.download(metadata, storage_dir)
+        logging.info(f"Downloading checkpoint {uuid} from Azure Blob Storage")
+        self.download(uuid, storage_dir)
 
         try:
-            yield os.path.join(self._base_path, metadata.storage_id)
+            yield os.path.join(self._base_path, uuid)
         finally:
-            self._remove_checkpoint_directory(metadata.storage_id)
+            self._remove_checkpoint_directory(uuid)
 
     @util.preserve_random_state
-    def upload(self, metadata: StorageMetadata, storage_dir: str) -> None:
-        for rel_path in metadata.resources.keys():
-            if not rel_path.endswith("/"):
-                rel_path_parent = (
-                    "{}/{}".format(metadata.storage_id, "/".join(rel_path.split("/")[:-1]))
-                ).rstrip("/")
-                container_name = "{}/{}".format(self.container, rel_path_parent)
-                blob_name = rel_path.split("/")[-1]
-                abs_path = os.path.join(storage_dir, rel_path)
-                logging.debug(
-                    "Uploading blob {} to container {}.".format(blob_name, container_name)
-                )
-                self.client.put(container_name, blob_name, abs_path)
-
-    @util.preserve_random_state
-    def download(self, metadata: StorageMetadata, storage_dir: str) -> None:
-        for rel_path in metadata.resources.keys():
+    def upload(self, uuid: str, storage_dir: str) -> None:
+        storage_prefix = uuid
+        for rel_path in sorted(self._list_directory(storage_dir)):
+            if rel_path.endswith("/"):
+                continue
+            # Use posixpath so that we always use forward slashes, even on Windows.
+            container_blob = posixpath.join(self.container, storage_prefix, rel_path)
+            blob_dir, blob_base = posixpath.split(container_blob)
             abs_path = os.path.join(storage_dir, rel_path)
+            logging.debug(f"Uploading blob {blob_base} to container {blob_dir}.")
+            self.client.put(blob_dir, blob_base, abs_path)
 
-            os.makedirs(os.path.dirname(abs_path), exist_ok=True)
+    @util.preserve_random_state
+    def download(self, uuid: str, storage_dir: str) -> None:
```

Comment:
how about calling `uuid` args throughout these functions `checkpoint_uuid`/`ckpt_uuid` instead?

I've experienced APIs where there're various types of objects both being identified with `uuid`s. e.g. hypothetical checkpoints and tasks where a task makes a checkpoint. all typed as strings. It's confusing to deduce which uuid you need to pass to some functions. having explicit name like `task_uuid` and `checkpoint_uuid` helps.

after all, the second argument to this method is `storage_dir`, not `dir`.

Reply:
Ok.  I actually changed these from being called `storage_id` everywhere to `uuid` everywhere, but your reasoning makes sense; I will switch back to `storage_id`.  (I'll do it in the `uuids` pr though, https://github.com/determined-ai/determined/pull/2970).